collection.go

```diff
@@ -49,6 +50,17 @@ func (cs *CollectionSpec) Copy() *CollectionSpec {
 	return &cpy
 }
 
+// ByteOrder returns the byte order of the Programs in the CollectionSpec.
+//
+// If the CollectionSpec doesn't contain any Programs, returns the host's
+// native endianness.
+func (cs *CollectionSpec) ByteOrder() binary.ByteOrder {
```

Comment:
I think I'd prefer adding a field `ByteOrder binary.ByteOrder` to CollectionSpec instead. What do you think?

Reply:
> adding a field `ByteOrder binary.ByteOrder` to CollectionSpec instead

I had it like this initially but it's a breaking API change for callers declaring `CollectionSpec` struct literals with implicit fields. (there were a few such spots in the codebase) It also needs to be included in `CollectionSpec.Copy()`, and it ultimately felt a bit fragile.

> the `internal.NativeEndian` fallback is weird

Ooops, this returns an internal value from an exported method. Unexporting is probably the way to go then. :+1:

Reply:
Looks like `internal.NativeEndian` is a package-global, not a const. I went with unexported `byteOrder binary.ByteOrder` instead. Knowing the byte order is required for parsing ELF map definitions as well, and this needs to work correctly for ELFs not containing any Programs as well.